FAQ.md

```diff
@@ -0,0 +1,257 @@
+
+To hide files in diffs, you can [mark them as generated](https://github.com/github/linguist#generated-code):
+
+```
+$ cat .gitattributes
+files/to/mark/as/generated/* linguist-generated
+```
+
+If you believe Linguist should already recognize these files as generated, you can submit a pull request to improve [our identification of generated files][`generated.rb`].
+
+
+### Why are some of my files not counted in the language statistics?
+
+Only files with a [markup or programming language](#what-are-markup-or-programming-languages) are counted in statistics. In addition, [generated][`generated.rb`], [documentation][`documentation.yml`], and [vendored][`vendor.yml`] files are excluded from statistics.
+
+
+### How can I trigger an update of the language detection in my repository?
+
+You can trigger a new analysis of your repository by pushing a change to your repository. This will enqueue a low priority background job that will analyze your repository. Keep in mind that it may take a while, particularly during busy periods, for your language statistics bar to reflect your changes.
+
+---
+
+## Syntax Highlighting
+
+### Why aren't my files syntax highlighted?
+
+There are three primary reasons why a file may not be highlighted:
+
+- the language for that file is not supported by Linguist, or
+- Linguist doesn't have a grammar to highlight files for that language or
+- Linguist was unable to properly detect the language.
+
+If [the language is not supported by Linguist](#how-can-i-check-if-linguist-supports-a-given-language), and you believe it meets [the requirements for support](#what-are-the-requirements-to-add-support-for-a-new-language-or-associate-a-new-extension-with-a-language), please consider [submitting a pull request](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language).
+
+You can see if Linguist has a grammar for the language in the [list of grammars][grammars]. If it doesn't and you know a Sublime Text, Atom or TextMate grammar that would work, please consider [submitting a pull request](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#fixing-syntax-highlighting).
+
+If Linguist supports the language and it has a grammar, the lack of syntax highlighting is probably the result of a misclassification. Please read [*The language detected for some files in my repository is incorrect*](#the-language-detected-for-some-files-in-my-repository-is-incorrect) to fix it.
+
+If a file was previously misclassified and you have implemented an override, you will need to push a change for the affected file too to invalidate the cached syntax highlighting.
+
+
+### How do I disable syntax highlighting for a file?
+
+You can disable syntax highlighting by telling Linguist the file is a Text file:
+
+```
+$ cat .gitattributes
+path/to/the/file linguist-language=Text
+```
+
+You will need to push a change for the affected file too to invalidate the cached syntax highlighting.
```

Comment:
Since this is always needed after a `.gitattributes` change, I'd rather keep it here than wait for the user to fall into this issue and go through the FAQ a second time.